Makefile

```diff
@@ -12,52 +12,91 @@
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
 
+# Deliverables path
 DESTDIR ?= /usr/local
-
-VERSION=$(shell git describe --match 'v[0-9]*' --dirty='.m' --always --tags)
+BINDIR ?= $(DESTDIR)/bin
+
+# Tools path
+RM ?= /bin/rm
+ECHO ?= /bin/echo
+DOCKER ?= /bin/docker
+GO ?= /bin/go
+MKDIR ?= /bin/mkdir
+TAR ?= /bin/tar
+INSTALL ?= /bin/install
+GIT ?= /bin/git
+
+VERSION ?= $(shell $(GIT) describe --match 'v[0-9]*' --dirty='.m' --always --tags)
 VERSION_TRIMMED := $(VERSION:v%=%)
-REVISION=$(shell git rev-parse HEAD)$(shell if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi)
+REVISION ?= $(shell $(GIT) rev-parse HEAD)$(shell if ! $(GIT) diff --no-ext-diff --quiet --exit-code; then $(ECHO) .m; fi)
 
 PKG_MAIN := github.com/containerd/fuse-overlayfs-snapshotter/cmd/containerd-fuse-overlayfs-grpc
 PKG_VERSION := github.com/containerd/fuse-overlayfs-snapshotter/cmd/containerd-fuse-overlayfs-grpc/version
 
-GO ?= go
 export GO_BUILD=GO111MODULE=on CGO_ENABLED=0 $(GO) build -ldflags "-s -w -X $(PKG_VERSION).Version=$(VERSION) -X $(PKG_VERSION).Revision=$(REVISION)"
 
-bin/containerd-fuse-overlayfs-grpc:
```

Comment:
Why is this changed?

Reply:
It's better give final binary name as target than using `make bin/containerd-fuse-overlayfs-grpc` to build target binary.

Reply:
Why?

Reply:
Ease of use, speaking generally `make <target-name>` is a standard practice. We don't use `make <path>/<target-name>`. Please let me know if you think otherwise here.

Reply:
Else we can keep this as it is and use `make binaries` or `make all` which will implicitly call this target.
Even `make containerd-fuse-overlayfs-grpc` is a lot of characters to type. :)

Reply:
The standard practice is to use file names as target names, and users are not expected to type the file names (target names) manually, as they are usually depended by the default target (such as `binaries`, `all`)

Reply:
Yeah, I guess we are good with this then ???

Reply:
The file name is `bin/containerd-fuse-overlayfs-grpc`, so the target name should be kept `bin/containerd-fuse-overlayfs-grpc`

Reply:
Okay